### PR TITLE
Refresh component list before retrieving metadata

### PR DESCRIPTION
**Pull request by brpowell**
### What does this PR do?

Refresh the component list before kicking off a multi-retrieve triggered by a metadata type BrowserNode. 

### What issues does this PR fix or reference?

W-6633366

### CODE REVIEW

File: packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveMetadata/describers.ts [hunk 1]
```diff
@@ -53,6 +53,7 @@ class TypeNodeDescriber extends NodeDescriber {
   }
 
   public async gatherOutputLocations(): Promise<LocalComponent[]> {
+    await OrgBrowser.get().refreshAndExpand(this.node);
```
Comment (brpowell) [inline packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveMetadata/describers.ts]: This updates the list and expands the metadata type node before attempting to gather LocalComponents from the children.

File: packages/salesforcedx-vscode-core/src/orgBrowser/browser.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { ExtensionContext, TreeView, window } from 'vscode';
+import { nls } from '../messages';
+import { BrowserNode, MetadataOutlineProvider } from '../orgBrowser';
+import { telemetryService } from '../telemetry';
+import { OrgAuthInfo } from '../util';
+
+export class OrgBrowser {
```
Comment (brpowell) [inline packages/salesforcedx-vscode-core/src/orgBrowser/browser.ts]: Needed to access the treeView and dataProvider outside of the `index.ts` file, so I created a separate container class to be an entry point for all Org Browser activity.

File: packages/salesforcedx-vscode-core/src/orgBrowser/index.ts [hunk 3]
```diff
@@ -8,3 +8,4 @@ export { TypeUtils, MetadataObject } from './metadataType';
 export { MetadataOutlineProvider } from './metadataOutlineProvider';
 export { BrowserNode, NodeType } from './nodeTypes';
 export { ComponentUtils } from './metadataCmp';
+export { OrgBrowser } from './browser';
```
Comment (lcampos) [inline packages/salesforcedx-vscode-core/src/orgBrowser/index.ts]: I think you can replace this with something like:
```
export const orgBrowser = OrgBrowser.get();
```
That way you don't have to do `OrgBrowser.get()` every time you import it in other places in the codebase e.g. line 408 in `packages/salesforcedx-vscode-core/src/index.ts`

File: docs/articles/user-guide/org-browser.md [hunk 4]
```diff
@@ -42,3 +42,7 @@ You can retrieve a component to your local project by clicking the retrieve butt
 You can also retrieve multiple components from the default org by clicking the retrieve button next to the metadata type. If retrieving a component overwrites it, you'll be prompted to select how to proceed.
 
 ![Overwrite components](../../images/overwrite-prompt.png)
+
+When the CLI retrieves components by a metadata type, the Org Browser needs an up-to-date list of all of the components to be retrieved. Therefore, the Org Browser will automatically refresh the component list before retrieving by a type. This is so the extensions can accurately check the local workspace for existing components.
```
Comment (brpowell) [inline docs/articles/user-guide/org-browser.md]: @hbc-bindu let me know if you need clarification with this new doc.

cc: @lcampos

Comment (hbc-bindu) [reply docs/articles/user-guide/org-browser.md]: @brpowell , reviewed the update. Let me know if the edits look ok.

File: packages/salesforcedx-vscode-core/test/vscode-integration/commands/forceSourceRetrieveMetadata/describers.test.ts [hunk 5]
```diff
@@ -9,13 +9,14 @@ import { expect } from 'chai';
 import { normalize } from 'path';
 import { sandbox, SinonStub } from 'sinon';
 import { RetrieveDescriberFactory } from '../../../../src/commands/forceSourceRetrieveMetadata';
-import { BrowserNode, NodeType } from '../../../../src/orgBrowser';
+import { BrowserNode, NodeType, OrgBrowser } from '../../../../src/orgBrowser';
```
Comment (lcampos) [inline packages/salesforcedx-vscode-core/test/vscode-integration/commands/forceSourceRetrieveMetadata/describers.test.ts]:
```suggestion
import { BrowserNode, NodeType, orgBrowser } from '../../../../src/orgBrowser';
```

File: packages/salesforcedx-vscode-core/test/vscode-integration/commands/forceSourceRetrieveMetadata/describers.test.ts [hunk 6]
```diff
@@ -28,8 +29,12 @@ describe('Retrieve Metadata Describers', () => {
   node.setComponents(['Test1', 'Test2', 'Test3'], NodeType.MetadataCmp);
 
   beforeEach(() => {
-    packageStub = env.stub(SfdxPackageDirectories, 'getPackageDirectoryPaths');
-    packageStub.returns(['p1', 'p2']);
+    packageStub = env
+      .stub(SfdxPackageDirectories, 'getPackageDirectoryPaths')
+      .returns(['p1', 'p2']);
+    refreshStub = env
+      .stub(OrgBrowser.prototype, 'refreshAndExpand')
```
Comment (lcampos) [inline packages/salesforcedx-vscode-core/test/vscode-integration/commands/forceSourceRetrieveMetadata/describers.test.ts]:
```suggestion
      .stub(orgBrowser.prototype, 'refreshAndExpand')
```

**Pull request merged**